dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -825,6 +846,32 @@ private GHRateLimit getGhRateLimitQuietly() {
         return startRateLimit;
     }
 
+    private GHRef[] getBranchesAndTags(GHRepository repo) throws IOException {
+        boolean getBranchesSucceeded = false;
+        boolean getTagsSucceeded = false;
+        GHRef[] branches = {};
+        GHRef[] tags = {};
+
+        // getRefs() fails if there are no matching results instead of returning an empty array/null.
+        try {
+            branches = repo.getRefs("refs/heads/");
+            getBranchesSucceeded = true;
+        } catch (IOException ex) {
+            LOG.debug("No branches found for " + repo.getName(), ex);
+        }
+        try {
+            tags = repo.getRefs("refs/tags/");
+            getTagsSucceeded = true;
+        } catch (IOException ex) {
+            LOG.debug("No tags found for " + repo.getName(), ex);
+        }
+
+        if (!getTagsSucceeded && !getBranchesSucceeded) {
```

Comment:
Suggest moving this into the 865/867 block, and throwing the original exception:

```
{ catch (IOException ex) {
LOG.debug(...)
if (!getBranchesSucceeded) {
   throw ex;
}
}
```